File: sdkwork-ai-sdk-flutter/lib/src/api/chat_completions.dart
import '../http/client.dart';
import '../models.dart';

class ChatCompletionsApi {
  final HttpClient _client;
  
  ChatCompletionsApi(this._client);

  /// Get chat completion
  Future<ChatCompletionResponse?> getChatCompletion(String completionId) async {
    final response = await _client.get(ApiPaths.aiPath('/management/chat/completions/${completionId}'));
    return response is ChatCompletionResponse ? response : null;
  }

  /// Update chat completion
  Future<ChatCompletionResponse?> updateChatCompletion(String completionId, UpdateChatCompletionRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/management/chat/completions/${completionId}'), body: body, contentType: 'application/json');
    return response is ChatCompletionResponse ? response : null;
  }

  /// Delete chat completion
  Future<ChatCompletionDeleteResponse?> deleteChatCompletion(String completionId) async {
    final response = await _client.delete(ApiPaths.aiPath('/management/chat/completions/${completionId}'));
    return response is ChatCompletionDeleteResponse ? response : null;
  }

  /// Get chat messages
  Future<ChatMessageList?> getChatMessages(String completionId, Map<String, dynamic>? params) async {
    final response = await _client.get(ApiPaths.aiPath('/management/chat/completions/${completionId}/messages'), params: params);
    return response is ChatMessageList ? response : null;
  }

  /// List chat completions
  Future<ChatCompletionList?> listChatCompletions(Map<String, dynamic>? params) async {
    final response = await _client.get(ApiPaths.aiPath('/management/chat/completions'), params: params);
    return response is ChatCompletionList ? response : null;
  }
}

File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/api/chat_completions.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import ChatCompletionDeleteResponse, ChatCompletionList, ChatCompletionResponse, ChatMessageList, UpdateChatCompletionRequest

class ChatCompletionsApi:
    """chat_completions API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def get_chat_completion(self, completionId: str) -> ChatCompletionResponse:
        """Get chat completion"""
        return self._client.get(f"/ai/v3/management/chat/completions/{completionId}")

    def update_chat_completion(self, completionId: str, body: UpdateChatCompletionRequest) -> ChatCompletionResponse:
        """Update chat completion"""
        return self._client.post(f"/ai/v3/management/chat/completions/{completionId}", json=body)

    def delete_chat_completion(self, completionId: str) -> ChatCompletionDeleteResponse:
        """Delete chat completion"""
        return self._client.delete(f"/ai/v3/management/chat/completions/{completionId}")

    def get_chat_messages(self, completionId: str, params: Optional[Dict[str, Any]] = None) -> ChatMessageList:
        """Get chat messages"""
        return self._client.get(f"/ai/v3/management/chat/completions/{completionId}/messages", params=params)

    def list_chat_completions(self, params: Optional[Dict[str, Any]] = None) -> ChatCompletionList:
        """List chat completions"""
        return self._client.get(f"/ai/v3/management/chat/completions", params=params)
